compiler/rustc_hir_analysis/src/check/mod.rs

```diff
@@ -546,3 +554,80 @@ fn bad_non_zero_sized_fields<'tcx>(
 pub fn potentially_plural_count(count: usize, word: &str) -> String {
     format!("{} {}{}", count, word, pluralize!(count))
 }
+
+pub fn check_function_signature<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    cause_code: ObligationCauseCode<'tcx>,
+    fn_id: LocalDefId,
+    expected_sig: ty::PolyFnSig<'tcx>,
```

Comment:
This should always just be a `FnSig`, is there ever a case where we're not passing `ty::Binder::dummy`?

Reply:
We use `ty::Binder::bind_with_vars` for intrinsics (I guess only needed for those with lifetime params) and for the panic handler.

Reply:
Yeah, ok